### PR TITLE
Adding patch api for registration.

### PR DESCRIPTION
**Pull request by ayush03agarwal**
Continuation of https://github.com/Flipkart/connekt/pull/270

### CODE REVIEW

File: receptors/src/main/scala/com/flipkart/connekt/receptors/routes/push/RegistrationRoute.scala [hunk 1]
```diff
@@ -61,17 +62,38 @@ class RegistrationRoute(implicit am: ActorMaterializer) extends BaseJsonHandler
...
+                            case Some(device) =>
+                              DeviceDetailsService.delete(appName, deviceId).get
+                              complete(GenericResponse(StatusCodes.OK.intValue, null, Response(s"DeviceDetails deleted for $deviceId", null)))
+                            case None =>
+                              complete(GenericResponse(StatusCodes.NotFound.intValue, null, Response(s"No Device Found for $appName / $deviceId", null)))
+                          }
+                        }
+                      } ~ patch {
+                        meteredResource(s"patch.$platform.$appName") {
+                          entity(as[Map[String, String]]) { patchedDevice =>
```
Comment (kingster) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/routes/push/RegistrationRoute.scala]: How is `DeviceDetails.keys` going to be updated ?

Comment (kingster) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/routes/push/RegistrationRoute.scala]: You need something else than Map[String, String]

Comment (ayush03agarwal) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/routes/push/RegistrationRoute.scala]: Actually I tried this, it worked fine with Map[String, String]
I can change Map[String, String] to Map[String, AnyRef]

Comment (ayush03agarwal) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/routes/push/RegistrationRoute.scala]: It was auto deserialising to Map[String, Map]

**Pull request merged**